lib/portage/dep/__init__.py

```diff
@@ -96,7 +97,11 @@ def _get_slot_dep_re(eapi_attrs):
     return slot_re
 
 
-def _match_slot(atom, pkg):
+def _match_slot(atom, pkg) -> bool:
+    """
+    @type atom: portage.dep.Atom
+    @type pkg: _emerge.Package.Package
```

Comment:
Do these annotations actually mean anything in terms of PEP 484?

Reply:
PEP 484 way of adding type hints to this function requires reordering. I thought the original ordering makes sense in a logical way. Shall I reorder the functions?

Reply:
If it's what's required to get proper typing working, I think so. I think it might be even better to pop out the classes from the `__init__` to their own dedicated file, for modularity. I'm not sure if there's an undesired side-effect that I'm not seeing, but it will help pre-empt these circular dependency issues (since you don't need all of the dependencies in `$CLASS` just to import the module the class is in).

Reply:
Even if we move the position of `_match_slot`, the annotation is not possible because of partial initialization of the module / circular imports. Like @ajakk said, we might need to move the class to a separate file. Can we merge this as it is and keep that for a separate pull request?

Reply:
OK, but please follow it up.